src/platform/silabs/efr32/Efr32PsaOperationalKeystore.h

```diff
@@ -94,11 +94,14 @@ class Efr32PsaOperationalKeystore : public chip::Crypto::OperationalKeystore
     bool mIsInitialized                      = false;
 
 private:
-    void ResetPendingKey()
+    void ResetPendingKey(bool keepOpaqueKey = false)
     {
         if (mPendingKeypair != nullptr)
         {
-            mPendingKeypair->Delete();
+            if (!keepOpaqueKey)
+            {
+                mPendingKeypair->Delete();
```

Comment:
This may need some comments explaining, since the `platform::Delete` seems to be done below anyway, so it seems unclear why delete may or may not be performed.

Maybe also a better name for `Delete` since I would expect a function named `Delete` to have already been called when `Platform::Delete(mPendingKeyPair)` is executed.

I would find it more intuitive to have the destructor always `Delete` but be able to call a `release` (or disown or similar) to read the code. So something like:

```
// ...
if (!keepOpaqueKey)
{
    mPendingKeyPair->ReleaseKey(); // Prevent delete from auto-freeing the key.
}
Platform::Delete(mPendingKeypair); // should automatically call Delete if the key is valid
```

Also signature wise a bool is generally not great. It may be easier to read if using some enum class: `ResetPendingKey(PendingKey::kReleaseBeforeDelete);`

Reply:
The `PendingKeypair->Delete();` is from the class `EFR32OpaqueKeypair` which will do a `psa_destroy_key` deleting the psa keypair from storage. In the context `keepOpaqueKey=true` is used we don't want to delete the key from storage has it is not committed (CommitOpKeypairForFabric). The key reference is kept in Cache and we only want to destruct the `mPendingKeypair` as it is not needed anymore.

I'll add some comments to make it clearer. I'll look into renames.

For the signature using enum class instead of a boolean; I guess I can do it, but I don't expect additional values or even for it to be used elsewhere. Since the end result is: destroy the psa keypair or not, I felt boolean was the correct choice. Maybe the naming can be better?